File: Bronze/568-SpeedingTicket/568.cpp
#include <iostream>

using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  int n, m;
  cin >> n >> m;

  int limit[100];
  int speed[100];
  int delta[100];
  int first = 0;

  for (int i = 0; i < n; i++) {
    int dist, slimit;
    cin >> dist >> slimit;
    for (int j = first; j < first + dist; j++) {
      limit[j] = slimit;
    }
    first = first + dist;
  }

  first = 0;

  for (int i = 0; i < m; i++) {
    int dist, sactual;
    cin >> dist >> sactual;
    for (int j = first; j < first + dist; j++) {
      speed[j] = sactual;
    }
    first = first + dist;
  }

  for (int a = 0; a < 100; a++) {
    delta[a] = speed[a] - limit[a];
  }

  cout << *max_element(delta, delta + 100) << "\n";
}

File: Bronze/568-SpeedingTicket/568.rs
use std::io;

fn main() {
    let mut n = String::new();
    let mut m = String::new();

    io::stdin().read_line(&mut n);
    io::stdin().read_line(&mut m);

    let n: i32 = n.trim().parse().expect("");
    let m: i32 = m.trim().parse().expect("");

    let mut limit: Vec<i32> = Vec::new();
    let mut speed: Vec<i32> = Vec::new();
    let mut delta: Vec<i32> = Vec::new();
    let mut first: i32 = 0;

    for i in (0..n) {
	let mut dist = String::new();
	let mut slimit = String::new();
	io::stdin().read_line(&mut dist);
	io::stdin().read_line(&mut slimit);
	let dist: i32 = dist.trim().parse().expect("");
	let slimit: i32 = slimit.trim().parse().expect("");
	let end = first + dist;
	for j in (first..end) {
	    limit.push(slimit);
	}
	first = first + dist;
    }

    first = 0;

    for i in (0..m) {
	let mut dist = String::new();
	let mut sactual = String::new();
	io::stdin().read_line(&mut dist);
	io::stdin().read_line(&mut sactual);
	let dist: i32 = dist.trim().parse().expect("");
	let sactual: i32 = sactual.trim().parse().expect("");
	let end = first + dist;
	for j in (first..end) {
	    speed.push(sactual);
	}
	first = first + dist;
    }

    for a in (0..100) {
	delta.push(&limit[a] - &speed[a]);
    }

    let max = delta.iter().max();
    match max {
	None => println!(""),
	Some(i) => println!("{}", "i")
    }
}
    
